### ISSUE TITLE
Constr param: Is there a way to preserve the regex annotation?

### ISSUE THREAD

**Issue by dwbelliston** [tiangolo/fastapi#2989]
Is there a way to preserve the regex annotation?

when i add constr the docs show just `str`

```
    use_filters: Optional[List[constr(regex=models.SEGMENT_PARAM_REGEX)]] = Query(
        None,
        alias="useFilters",
        title="Use Filters",
        # regex=models.SEGMENT_PARAM_REGEX,
        description="Filter results using Field:Operator:Value",
    ),
```    

![image](https://user-images.githubusercontent.com/11450118/109264009-804f3480-77c1-11eb-9a9f-c110aa16b436.png)

but with the regex on Query it shows up in the docs

```
    use_filters: Optional[List[str]] = Query(
        None,
        alias="useFilters",
        title="Use Filters",
        regex=models.SEGMENT_PARAM_REGEX,
        description="Filter results using Field:Operator:Value",
    ),
```

![image](https://user-images.githubusercontent.com/11450118/109264097-ad034c00-77c1-11eb-8bbf-e9d30eba4ae0.png)

_Originally posted by @dwbelliston in https://github.com/tiangolo/fastapi/issues/1021#issuecomment-786443902_

**Comment by tiangolo** [comment]
Yep, exactly that, you should put the regex in the parameter passed to `Query` instead of using the `const` function. Both achieve the same.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
